spool.py

```diff
@@ -113,43 +117,52 @@ def spool_render(rman_version_short, rib_files, denoise_files, denoise_aov_files
+            files = [" ".join(denoise_aov_files[frame_num -
+                                                frame_begin])] if rm.spool_denoise_aov else []
+            cmd_str = ['denoise'] + denoise_options + \
+                [denoise_files[frame_num - frame_begin][0]] + files
             write_cmd_task_line(f, 'Denoise frame %d' % frame_num,
                                 [('PixarRender', cmd_str)], 3)
         elif crossframe_denoise:
-            denoise_options = ['--crossframe -v variance ' + rm.denoise_cmd] if rm.denoise_cmd != '' else ['--crossframe -v variance']
+            denoise_options = ['--crossframe -v variance ' +
+                               rm.denoise_cmd] if rm.denoise_cmd != '' else ['--crossframe -v variance']
             if rm.spool_denoise_aov and denoise_aov_files != []:
                 denoise_options.append('--filtervariance 1')
             if frame_num - frame_begin < 1:
                 pass
             elif frame_num - frame_begin == 1:
                 denoise_options.append('-L 1')
-                cmd_str = ['denoise'] + denoise_options + [f[0] for f in denoise_files[0:2]]
-                if rm.spool_denoise_aov:
-                    files = [item for sublist in denoise_aov_files[0:2] for item in sublist]
-                    cmd_str = ['denoise'] + denoise_options + [f[0] for f in denoise_files[0:2]] + files
+                files = [item for sublist in denoise_aov_files[0:2]
+                         for item in sublist] if rm.spool_denoise_aov else []
+                cmd_str = ['denoise'] + denoise_options + [f[0]
+                                                           for f in denoise_files[0:2]] + files
                 write_cmd_task_line(f, 'Denoise frame %d' % (frame_num - 1),
                                     [('PixarRender', cmd_str)], 3)
             else:
                 denoise_options.append('-F 1 -L 1')
-                cmd_str = ['denoise'] + denoise_options + [f[0] for f in denoise_files[frame_num - 3: frame_num]]
-                if rm.spool_denoise_aov:
-                    files = [item for sublist in denoise_aov_files[frame_num - 3: frame_num] for item in sublist]
-                    cmd_str = ['denoise'] + denoise_options + [f[0] for f in denoise_files[frame_num - 3: frame_num]] + files
+                files = [item for sublist in denoise_aov_files[frame_num - frame_begin -
+                                                               2: frame_num - frame_begin + 1] for item in sublist] if rm.spool_denoise_aov else []
+                cmd_str = ['denoise'] + denoise_options + [f[0] for f in denoise_files[
+                    frame_num - frame_begin - 2: frame_num - frame_begin + 1]] + files
                 write_cmd_task_line(f, 'Denoise frame %d' % (frame_num - 1),
                                     [('PixarRender', cmd_str)], 3)
             if frame_num == frame_end:
-                denoise_options.remove('-F 1 -L 1')
+                try:
```

Comment:
Hey one more thing:  try/excepts shouldn't be used here, instead do:

if ('-F 1 -L 1') in denoise_options:
     denoise_options.remove(...)